packages/react-input/Spec.md

```diff
@@ -96,13 +96,18 @@ type SimplifiedInputProps = {
 
   /** Called when the user changes the value */
   onChange?: (ev: React.FormEvent<HTMLInputElement>, data: { value: string }) => void;
+
+  /** Allowed values for the native `type` prop */
+  type?: 'text' | '...'; // this is an enumeration of all text-like values
 };
 ```
 
 Notes on native prop conflicts/overrides:
 
 - `size` [overlaps with a native prop](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/size) which sets the width of the field in "number of characters." This isn't ideal, but we're going with it since the native prop isn't very useful in practice, and it was hard to find another reasonable/consistent name for the visual size prop. It's also consistent with the approach used by most other libraries which have a prop for setting the visual size. (If anyone needs the native functionality, we could add an `htmlSize` prop in the future.)
 - `value` and `defaultValue` are defined in `InputHTMLAttributes` (from `@types/react`) as `string | ReadonlyArray<string> | number` since the same props interface is used for all input element types. To reflect actual usage, we override the types to only accept strings.
+- `onChange` is overridden per the [RFC on event handler arguments](https://github.com/microsoft/fluentui/blob/master/rfcs/convergence/event-handlers-arguments.md).
+- `type` by default has type `string`, but it's restricted to a list of only the text-like values [listed on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#input_types). I think it's worth making this explicit to avoid people thinking that Input can handle all the same behaviors as a native `<input>`.
```

Comment:
Nit: this reads a little strange to me. Are you referring to the TS type `string` or is this a typo that should be "...has type `text`"? Isn't the TS type always string for this prop?

Reply:
That would be a typo, good catch

Reply:
Oh actually I misread my text/your comment. I was referring to the TS type of that property from `@types/react` which is `string`.

Reply:
Updated with wording that's hopefully clearer